ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HeaderValidation.hs

```diff
@@ -499,6 +506,50 @@ deriving instance StandardHash blk => Eq   (TipInfoIsEBB blk)
 deriving instance StandardHash blk => Show (TipInfoIsEBB blk)
 deriving instance StandardHash blk => NoThunks (TipInfoIsEBB blk)
 
+{-------------------------------------------------------------------------------
+  Header with time
+-------------------------------------------------------------------------------}
+
+-- | A header and the slot time at which it was received
+--
+-- REVIEW: we originally talked about calling this time
+-- 'ValidatedHeader' however this seems to be a misnomer since the
+-- header is not validated at the point at which we construct this
+-- value in 'Ouroboros.Consensus.MiniProtocol.ChainSync.Client'.
```

Comment:
I think that depends on where you construct `HeaderWithTime`/`ValidatedHeader`.

(Generally, `HeaderWithTime` sounds good though :+1:)